# Snakemake workflow for variant calling from RNA-seq data

This Snakemake workflow implements the [GATK best-practices workflow for RNA-seq short variant discovery](https://gatk.broadinstitute.org/hc/en-us/articles/360035531192-RNAseq-short-variant-discovery-SNPs-Indels).

## Usage

Standardised usage of this workflow is described in the [Snakemake Workflow Catalog](https://snakemake.github.io/snakemake-workflow-catalog/?usage=baerlachlan/smk-rnaseq-gatk-variants).

However, if the intention is to run this workflow in an HPC environment where internet access is not be available on compute nodes, downloading the workflow from [Releases](https://github.com/baerlachlan/smk-rnaseq-gatk-variants/releases) is recommended.
